lib/src/stream_sink_extensions.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'stream_sink_transformer.dart';
+import 'stream_sink_transformer/reject_errors.dart';
+
+/// Extensions on [StreamSink] to make stream transformations more fluent.
+extension StreamSinkExtensions<T> on StreamSink<T> {
+  /// Transforms a [StreamSink] using [transformer].
+  StreamSink<S> transform<S>(StreamSinkTransformer<S, T> transformer) =>
+      transformer.bind(this);
+
+  /// Returns a [StreamSink] that forwards to [this], but doesn't allow any
+  /// errors to be passed in.
```

Comment:
First line should be shorter.
The `[this]` is not a valid doc link (not an identifier, not a source declaration).
Generally, never start a dartdoc with "Returns".

Maybe:
> /// Transforms this stream sink into one which rejects errors.
> ///
> /// Returns a new [StreamSink]. Any non-error event added to that new sink,
> /// whether by [add], [close] or [addStream], is also added to this stream sink.
> /// Any error event added to the new sink, whether by [addError] or [addStream],
> /// will instead close this sinks, the error becomes the result of the returned
> /// sink's [StreamSink.done] future, and further events added to the new sink
> /// are ignored.

Reply:
> First line should be shorter.

Done.

> The `[this]` is not a valid doc link (not an identifier, not a source declaration).

I don't think this is accurate—the SDK has over 100 instances of `[this]`, and it does seem to be linked by dartdoc (see [this example](https://api.dart.dev/stable/2.12.4/dart-typed_data/Float32x4/scale.html)).

> Generally, never start a dartdoc with "Returns".

That's contrary to [Effective Dart](https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-function-or-method-comments-with-third-person-verbs)'s suggestion.

Reply:
TIL that `[this]` is valid and links to the class itself. I wish DartDoc had, well, documentation (I still don't remember how to link to an unnamed constructor).

I disagree with the Effective Dart *example* here. The suggestion to use active verbs is fine, it's just that the word "Returns" is always inferior to some other word (or none, for getter-like functions where you should use a noun phrase).

If the *only* or *most important* thing you can say about a function is which object it returns, it should be treated as a getter-like function ("parameterized getter") and documented as such (noun phrase).

Most of the time, it's not the most important thing, and the *operation* performed by the function is more relevant. In this case, "Creates" is a better word since it emphasized that the returned sink is *new*.

So, it's not that "Returns" is technically wrong, I've just found that it's just *never* the *best* word to use. Seeing it means that the text can be improved. 


(Having a later section starting with "Returns ..." is highly recommended as the way to describe the returned value. Users should be trained to look for such a section. That too gets muddled when the first overview sentence starts with "Returns" as well).